flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/Dispatcher.java

```diff
@@ -1251,7 +1253,20 @@ private CompletableFuture<Void> removeJob(JobID jobId, CleanupJobState cleanupJo
         if (cleanupJobState.isGlobalCleanup()) {
             return globalResourceCleaner
                     .cleanupAsync(jobId)
-                    .thenRunAsync(() -> markJobAsClean(jobId), ioExecutor)
+                    .thenRunAsync(
```

Comment:
You could use `thenComposeAsync` here to avoid the `.get()` call. I'm wondering why we're using `Async here instead of just `thenRun` :thinking:

Reply:
Fixed.